dotcom-rendering/src/web/layouts/FrontLayout.tsx

```diff
@@ -34,26 +35,91 @@ const spaces = / /g;
 const ophanComponentId = (name: string) =>
 	name.toLowerCase().replace(spaces, '-');
 
-const shouldInsertMerchHigh = (
-	index: number,
-	isNetworkFront: boolean | undefined,
+const getMerchHighPosition = (
 	collectionCount: number,
-	isPaidContent: boolean | undefined,
+	isNetworkFront: boolean | undefined,
 ) => {
-	const minContainers = isPaidContent ? 1 : 2;
-	if (collectionCount < minContainers) {
-		return false;
-	}
-	let desiredPosition;
 	if (collectionCount < 4) {
-		desiredPosition = 2;
+		return 2;
 	} else if (isNetworkFront) {
-		desiredPosition = 5;
+		return 5;
 	} else {
-		desiredPosition = 4;
+		return 4;
 	}
+};
+
+// On mobile, we remove the first container if it is a thrasher
+// and remove a container if it, or the next sibling, is a commercial container
+// we also exclude any containers that are directly before a thrasher
+// then we take every other container, up to a maximum of 10, for targeting MPU insertion
+
+const getMobileAdPositions = (
+	isNetworkFront: boolean | undefined,
+	collections: DCRCollectionType[],
+) => {
+	const positions: number[] = [];
+	const merchHighPosition = getMerchHighPosition(
+		collections.length,
+		isNetworkFront,
+	);
+
+	collections.forEach((collection, collectionIndex) => {
+		const isThrasher = collection.collectionType === 'fixed/thrasher';
+		const isFirst = collectionIndex === 0;
+		const isNearMerchandising =
+			collectionIndex === merchHighPosition ||
+			collectionIndex + 1 === merchHighPosition;
+		const isNearThrasher =
+			collections[collectionIndex + 1]?.collectionType ===
+			'fixed/thrasher';
+		if (isFirst && isThrasher) return false;
+		if (isNearMerchandising) return false;
+		if (isNearThrasher) return false;
+		else if (
+			collectionIndex % 2 === 0 &&
+			collectionIndex < collections.length - 1
+		) {
+			positions.push(collectionIndex);
+		}
+		return false;
+	});
```

Comment:
What followes is a code style suggestion you may want to ignore!

If you are keen to use a purely functional style without side-effects, you could use `reduce`, or a series of `map` & `filter`. This is not a major issue, as the side effects do not escape the scope of this function, but happy to offer support if you would like to refactor now or at some later point.

For example:

```ts
const positions = collections
  .map((collection, collectionIndex) => {
    const isThrasher = collection.collectionType === "fixed/thrasher";
    const isFirst = collectionIndex === 0;
    const isNearMerchandising =
      collectionIndex === merchHighPosition ||
      collectionIndex + 1 === merchHighPosition;
    const isNearThrasher =
      collections[collectionIndex + 1]?.collectionType === "fixed/thrasher";
    if (isFirst && isThrasher) return false;
    if (isNearMerchandising) return false;
    if (isNearThrasher) return false;
    else if (
      collectionIndex % 2 === 0 &&
      collectionIndex < collections.length - 1
    ) {
      // This is where we avoid the mutation
      return true;
    }
    return false;
  })
  .map((shouldDisplayAd, collectionIndex) =>
    shouldDisplayAd ? collectionIndex : undefined
  )
  // we filter out `undefined` values
  .filter((index) => typeof index === "number")
  .slice(0, 10);
```